sys/net/application_layer/dhcpv6/client.c

```diff
@@ -131,20 +149,47 @@ static void *_thread(void *args)
     event_queue_t event_queue;
     event_queue_init(&event_queue);
     dhcpv6_client_init(&event_queue, SOCK_ADDR_ANY_NETIF);
-    /* TODO: add configuration for IA_NA here, when implemented */
     dhcpv6_client_start();
     event_loop(&event_queue);   /* never returns */
     return NULL;
 }
 #endif /* MODULE_AUTO_INIT_DHCPV6_CLIENT */
 
+void _initialize_ia_na(uint16_t netif)
+{
+    if (!IS_USED(MODULE_DHCPV6_CLIENT_IA_NA)) {
+        return;
+    }
+
+    /* If no specific interface ID is given, check all
+       interfaces if DHCP IA_NA is enabled. Otherwise
+       use the specific interface ID. */
+    if (netif == SOCK_ADDR_ANY_NETIF) {
+        netif_t* netif = NULL;
+        while ((netif = netif_iter(netif))) {
+            int16_t netif_id = netif_get_id(netif);
+            if (netif_id < 0) {
+                continue;
+            }
+            if (dhcpv6_client_check_ia_na(netif_id)) {
+                dhcpv6_client_req_ia_na(netif_id);
```

Comment:
... and then bail out here accordingly.

Reply:
Would printing a debug message be enough if `dhcpv6_client_req_ia_na` returns a value < 0? Or should there be some kind of assertion or error handling?

Reply:
Yepp, please only add a DEBUG statement or LOG message here. We don't want the node to crash only because not enough memory was allocated.

Reply:
I added a debug message in b7701e17a77087957ffb0c6b5f7e105f86a3d471. I didn't cover the `-ENOTSUP` case yet, however, as there is already a `LOG_WARNING` in `dhcpv6_client_req_ia_na`. If it makes sense to also print a debug message in this case I could add this of course.